log/term/colorlogger.go

```diff
@@ -0,0 +1,149 @@
+package term
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+)
+
+// Color is the abstract color, the zero value is the Default.
+type Color uint8
+
+const (
+	NoColor = Color(iota)
+	Red
+	Green
+	Yellow
+	Blue
+	Magenta
+	Cyan
+	White
+	Default
+
+	maxColor
+)
+
+var resetColorBytes = []byte("\x1b[0m")
+var fgColorBytes [][]byte
+var bgColorBytes [][]byte
+
+func init() {
+	for color := NoColor; color < maxColor; color++ {
+		fgColorBytes = append(fgColorBytes, []byte(fmt.Sprintf("\x1b[%dm", 30+color)))
+		bgColorBytes = append(bgColorBytes, []byte(fmt.Sprintf("\x1b[%dm", 40+color)))
+	}
+}
+
+type FgBgColor struct {
+	Fg, Bg Color
+}
+
+func (c FgBgColor) IsZero() bool {
+	return c.Fg == NoColor && c.Bg == NoColor
+}
+
+// NewColorLogger returns a log.Logger which writes colored logs to w. It
+// colors whole records based on the FgBgColor returned by the color function.
+// Log events are formatted by the Logger returned by newLogger.
+func NewColorLogger(w io.Writer, newLogger func(io.Writer) log.Logger, color func(keyvals ...interface{}) FgBgColor) log.Logger {
+	if color == nil {
+		panic("color func nil")
+	}
```

Comment:
I respect your opinion on this, but I also feel that the emergent philosophy of Go kit aligns with the Unix philosophy which pushes the other direction. I propose we keep `ColorLogger` as it is with the semantics of *write colored log events to an `io.Writer`*, and provide a higher level convenience that combines the `IsTty` check, `ColorLogger`, and a vendored go-colorable to provide the convenience you want. I think having all of these in the log/term package makes sense. The main package documentation can focus on the most convenient approach but the building blocks are still accessible to anyone with more specialized needs.

I imagine the package would export:
- `TtyLogger`  (the all-in-one convenience)
- `ColorLogger`
- `TtyColorWriter` (go-colorable logic as an `io.Writer` decorator)
- `IsTty`